cmd/rpcdaemon/commands/eth_call.go

```diff
@@ -106,7 +106,7 @@ func (api *APIImpl) DoEstimateGas(ctx context.Context, args ethapi.CallArgs, blo
 		}
 	}
 	// Recap the highest gas allowance with specified gascap.
-	if gasCap != nil && hi > gasCap.Uint64() {
+	if gasCap != nil && gasCap.Uint64() != 0 && hi > gasCap.Uint64() {
```

Comment:
One tiny nitpick that there should be IsZero() method on gasCap

Reply:
It gives me error when I use `gasCap.IsZero()`: `gasCap.IsZero undefined (type *big.Int has no field or method IsZero)`

Reply:
I with with the `*big.Int` the way to check for zero is `gasCap.Sign() == 0`, that works regardless of whether it fits into uint64 or not

Reply:
So for checking I than used gasCap.Sign() > 0, so in case o zero or negative value it would return false.

Reply:
Thank you! I will do a quick test a bit later and merge if it is ok